finetune/input_pipeline.py

```diff
@@ -423,6 +418,26 @@ def get_train_input_fns(self, Xs, Y=None, batch_size=None, val_size=None, contex
             .cache()
             .prefetch(prefetch_buffer)
         )
+
+        if self.config.mlm_context_shuffle_val and context is not None:
+            
+            def shuffle_context(x):
+                context = tf.random.shuffle(x["context"])
+                x["context"] = context
+                return x
+            
+            shuffled_context_val = (
+                lambda: val_dataset_unbatched()
+                .map(shuffle_context)
+                .padded_batch(batch_size, padded_shapes=shapes, drop_remainder=False)
+                .cache()
+                .prefetch(prefetch_buffer)
+            )
+            val_dataset = {
```

Comment:
I'm confused as to what dark magic is happening here.

Reply:
So I modified some of the api used to do multiple validation runs for MTL... Don't tell anyone

Reply:
Mostly the `None: val_dataset` portion.

Reply:
Ahh, yes. So the InMemoryEvaluator hook gives it the default name if value that goes in is None, it needs the default name for keep best model to work.

Reply:
Definitely not a permanent fixture.

Reply:
Got it. Thank you